app/models/miq_expression.rb

```diff
@@ -1593,6 +1593,31 @@ def self._quick_search?(e)
     end
   end
 
+  def fields
+    self.class._fields(exp)
+  end
+
+  def self._fields(expression)
+    result = []
+    case expression
+    when Array
+      expression.each { |x| result.concat(_fields(x)) }
+    when Hash
+      if expression.keys.include?('field')
```

Comment:
If you can be certain that you have a `field` here, I wonder if you could parse it with `Field.parse` and have `Field` tell you if it has a custom_attribute column?